src/extension.ts

```diff
@@ -151,12 +151,12 @@ async function startLanguageServer() {
     const serverArgsDebug = ['--startup-file=no', '--history-file=no', '--depwarn=no', `--project=${envForLSPath}`, 'main.jl', jlEnvPath, '--debug=yes', telemetry.getCrashReportingPipename(), oldDepotPath, g_context.globalStoragePath]
     const spawnOptions = {
         cwd: path.join(g_context.extensionPath, 'scripts', 'languageserver'),
-        env: {
+        env: Object.assign(process.env, {

```

Comment:
I'm quite nervous about this one, to be honest... One of the key design decisions around LS was that we isolate it completely from user configuration settings that might hang around on a system, and with this one here now we are pulling _everything_ in that a user might have configured in terms of env vars for every user, which I think we should not do. One of the reasons why we haven't had many problems about configuration is that we have been so strict about that.

Could we do a more targeted approach here, or something that is more configurable? For example, we could have an configuration option `advancedLSenv`, which is a string, empty be default, that gets added here. Or we have a bool configuration option `advancedPassSystemEnvToLS` that is `false` by default, or something like that.

Reply:
@davidanthoff  thanks for the observation. I am not familiar with your code base, including any of the mechanism to drive it. I find it useful and performing well -- with the exception of **silent crashes** with other engineered solutions such as Intel MKL. These modifications including the `:` colon with Julia Depot fixes my problems -- also passes ALL TEST cases you have on your CI.

As a consumer to this open sourced excellent work I uploaded the diffs, with good intentions: to help others from silent crashes. The change was only possible as @pfitzseb  [showed me](https://discourse.julialang.org/t/vscode-julia-doesnt-find-intel-mkl-shared-objects/52232/8?u=steven_varga) how to compile these packages -- as I am not familiar with the framework/workflow. In my opinion, more qualified people such as you and @pfitzseb should use it or not as you wish. 

In a [different thread](https://github.com/julia-vscode/julia-vscode/pull/1839/files/9b4e3b34b9fc8159a2a4bdc077cadc2e923ad868#r552692845) @pfitzseb asked to remove the addition of ':' based on similar argument. Combining the two modifications, we got zero line.

Reply:
As below, this *should* be a no-op. So we could just run with it if it actually fixes the MKL issue.